src/sage/features/__init__.py

```diff
@@ -921,14 +921,14 @@ def _is_present(self):
                 from distutils.errors import CCompilerError
             except ImportError:
                 CCompilerError = ()
-        with open(tmp_filename(ext='.pyx'), 'w') as pyx:
-            pyx.write(self.test_code)
         try:
             from sage.misc.cython import cython_import
         except ImportError:
             return FeatureTestResult(self, False, reason="sage.misc.cython is not available")
         try:
-            cython_import(pyx.name, verbose=-1)
+            with TemporaryFile(suffix='.pyx') as pyx:
+                pyx.write(self.test_code)
+                cython_import(pyx.name, verbose=-1)
```

Comment:
I don't think this will work on POSIX systems. The docs say,

> The returned object is a true file object on POSIX platforms. On other platforms, it is a file-like object whose file attribute is the underlying true file object.

What it does on my machine:

```
>>> import tempfile
>>> with tempfile.TemporaryFile() as f:
...     print(f.name)
... 
3
```

Reply:
Right, it should have used `NamedTemporaryFile`. Strange thing is that on CI the Cython features `bliss` and `sage.misc.cython` were nevertheless correctly recognized.